src/components/EtherscanLink.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { ReactElement } from 'react'
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import { abbreviateString } from 'utils'
+import { Network } from 'types'
+
+export enum EtherscanLinkType {
```

Comment:
nit: maybe is simpler just to use a type like `'tx' | 'address' | ...`, so is still checked, but easier to use:
* Don't need to import the type, just the component
* less verbose to use
* The IDE still does the check and you even get a suggestion of posible values

Reply:
Addressed in the latest revision :)